### ISSUE TITLE
On the Distinction Between `loobean` and `boolean`

### ISSUE THREAD

**Issue by DavePearce** [Consensys/corset#189]
This is a feature request from @OlivierBBB to reconsider whether it is useful to distinguish between `loobean` and `boolean`.  The argument against distinguishing them is roughly that having duplicate primitives, such as a `eq` vs `eq!` and `or` vs `or!` is awkward and can lead to confusion.  I want to think through the implications here so we understand the tradeoffs.

## Summary

The reason for distinguishing between `loobean` and `boolean` is that logic is interpreted different depending upon its context.  An obvious situation where this arises is with guards, which are adopt `boolean` semantics (i.e. because they are typically columns).

## Examples

Example where `eq` is used (there is only one occurrence in fact):
```
(defcolumns STAMP CT CT_MAX)

(defconstraint test ()
  (if-not-zero STAMP
        (if (eq CT CT_MAX)
            (will-inc! STAMP 1))))
```

## Notes

* No columns marked as `loob` or `bool`.
* `and!` and `or` are never used.  However, `and` and `or!` are used.
* `eq` is only used in one place, whilst `eq!` is used in many places.

**Comment by delehef** [comment]
Those were mostly tentatives to introduce type-safe, compiler-checked, higher-level, building blocks, but it turns out that implementors preferred to fall back on primitives such as multiplications & subtraction, specific `if-eq`, etc.

So if they are mostly a support burden without application, I reckon dropping them is not the worst idea.

**Comment by DavePearce** [comment]
> Those were mostly tentatives to introduce type-safe, compiler-checked, higher-level, building blocks

Yes, I understand what you were trying to do here.  It would be nice to see things move more towards being a "regular" programming language!!  I can see now that that is definitely possible.

Anyway, thanks for the input.  What I'm hearing is that there is nothing really major I didn't know about which would block this.
